src/components/VersionsBar/VersionsBar.tsx

```diff
@@ -119,54 +123,103 @@ export function VersionsBar({preparedVersions}: VersionsBarProps) {
                                 handleMouseEnter(item.version);
                             }}
                             onMouseLeave={handleMouseLeave}
-                            className={b('version', {dimmed: isDimmed(item.version)})}
+                            className={b('version', {dimmed: isDimmed(item.version), size})}
                             style={{backgroundColor: item.color, width: `${item.value}%`}}
                         />
                     </Tooltip>
                 ))}
             </Flex>
 
-            <Flex gap={0.5} direction={'column'}>
-                {truncatedDisplayedVersions.map((item) => (
-                    <Tooltip
-                        key={item.version}
-                        content={i18n('tooltip_nodes', {count: item.count})}
-                        placement={'bottom-end'}
-                        openDelay={TOOLTIP_OPEN_DELAY}
-                    >
-                        <Flex gap={1} alignItems={'center'} className={b('titles-wrapper')}>
-                            <svg
-                                xmlns="http://www.w3.org/2000/svg"
-                                width="6"
-                                height="6"
-                                viewBox="0 0 6 6"
-                                fill="none"
-                                className={b('version-icon', {dimmed: isDimmed(item.version)})}
-                            >
-                                <circle cx="3" cy="3" r="3" fill={item.color} />
-                            </svg>
-                            <div
-                                className={b('title', {dimmed: isDimmed(item.version)})}
-                                onMouseEnter={() => {
-                                    handleMouseEnter(item.version);
-                                }}
-                                onMouseLeave={handleMouseLeave}
-                            >
-                                {item.version}
-                            </div>
-                        </Flex>
-                    </Tooltip>
-                ))}
-                <Flex>{renderButton()}</Flex>
-            </Flex>
+            {withTitles && (
+                <Flex className={b('titles-wrapper', {size})}>
+                    {truncatedDisplayedVersions.map((item) => (
+                        <VersionTitle
+                            key={item.version}
+                            version={item.version}
+                            color={item.color || ''}
+                            count={item.count}
+                            isDimmed={isDimmed(item.version)}
+                            onMouseEnter={() => {
+                                handleMouseEnter(item.version);
+                            }}
+                            onMouseLeave={handleMouseLeave}
+                            size={size}
+                        />
+                    ))}
+                    <Flex>{renderButton()}</Flex>
```

Comment:
renderButton renders one element - why do we need Flex here?

Reply:
It's needed to limit button width. Replaced it with css `width: max-content`

<img width="378" height="121" alt="Screenshot 2025-08-15 at 16 28 50" src="https://github.com/user-attachments/assets/ad869954-af21-4eab-a01b-c0f3760734ad" />